ssg/yaml.py

```diff
@@ -121,7 +125,7 @@ def open_raw(yaml_file):
     return yaml_contents
 
 
-def ordered_load(stream, Loader=yaml.Loader, object_pairs_hook=OrderedDict):
+def ordered_load(stream, Loader=yaml_SafeLoader, object_pairs_hook=OrderedDict):
```

Comment:
So there is the "suspicious" line 33 which modifies `yaml_SafeLoader`:

```
yaml_SafeLoader.add_constructor(u'tag:yaml.org,2002:bool', _bool_constructor)
```

But before your patch the `yaml_SafeLoader` was used only in the `_open_yaml` function. So this special modfication from line 33 affected only the  `_open_yaml` function. But, you start to use the `yaml_SafeLoader` also here in `ordered_load`, so now this special modification affects also the `ordered_load` function.

If we would like to use `yaml_SafeLoader` in both functions but keep the previous behavior of `ordered_load`, we would have to rework it so that the special modification from line 33 would again affect only `_open_yaml`. We can do that using a trick like this:

```
diff --git a/ssg/yaml.py b/ssg/yaml.py
index 71f7e3adee..601584a21d 100644
--- a/ssg/yaml.py
+++ b/ssg/yaml.py
@@ -29,8 +29,6 @@ def _unicode_constructor(self, node):
     return str(string_like)
 
 
-# Don't follow python bool case
-yaml_SafeLoader.add_constructor(u'tag:yaml.org,2002:bool', _bool_constructor)
 # Python2-relevant - become able to resolve "unicode strings"
 yaml_SafeLoader.add_constructor(u'tag:yaml.org,2002:python/unicode', _unicode_constructor)
 
@@ -52,8 +50,13 @@ def _open_yaml(stream, original_file=None, substitutions_dict={}):
 
     Return None if it contains "documentation_complete" key set to "false".
     """
+    class OurCustomLoader(yaml_SafeLoader):
+        pass
+
+    # Don't follow python bool case
+    OurCustomLoader.add_constructor(u'tag:yaml.org,2002:bool', _bool_constructor)
     try:
-        yaml_contents = yaml.load(stream, Loader=yaml_SafeLoader)
+        yaml_contents = yaml.load(stream, Loader=OurCustomLoader)
 
         if yaml_contents.pop("documentation_complete", "true") == "false" and \
                 substitutions_dict.get("cmake_build_type") != "Debug":
```

I admit that I don't like that we have some special things like this `yaml_SafeLoader.add_constructor(u'tag:yaml.org,2002:bool', _bool_constructor)`, it would be ideal to remove it, but that would be very very difficult because a lot of code depend on this.